### DISCUSSION TITLE
Request for help on formatting and presenting the result from REST API

### DISCUSSION THREAD

**Discussion by quantsegu** [Budibase/budibase#12839] [Help] [answered]
Hi guru's

I am sending a REST API Query based on parameters and receiving a response. But the response is full JSON,i only want to show the one value in my output but in a card not in a table. 

So i am getting user input required for Post, and then user submits the request. Then the request is sent through actions assigned to button.

First action, execute query with variables
second action, fetch data and assign it to state
use state to present in the UI

Is there a way i can format and present the outpt for only one key in the JSON file.

JSON file looks like following:

```javascript
{"data":[{"value":"SELECT AVG(column_name) FROM table_name;"}],"pagination":{"cursor":null},"headers":{"date":"Mon, 22 Jan 2024 17:32:57 GMT","content-type":"application/json","transfer-encoding":"chunked","connection":"close","access-control-allow-origin":"*","cache-control":"no-cache, must-revalidate","openai-model":"gpt-4-0613","openai-organization":"user-ewfwgtb07y6lq7mpmr8q","openai-processing-ms":"520","openai-version":"2020-10-01","strict-transport-security":"max-age=15724800; includeSubDomains","x-ratelimit-limit-requests":"10000","x-ratelimit-limit-tokens":"0","x-ratelimit-remaining-requests":"9999","x-ratelimit-remaining-tokens":"9701","x-ratelimit-reset-requests":"8.64s","x-ratelimit-reset-tokens":"1.794s","x-request-id":"c7be58bf0ba5a77c4c3b2573242bd9dc","cf-cache-status":"DYNAMIC","set-cookie":"__cf_bm=c4zDa9vyMw8MGVUdpBlNQ0u34MjrBlHSKKKSwoGSG1s-1705944777-1-ARTHrT+ZET9bbrsNf5HrqRleDiXsFCRvPmf6kw6ZQQDkuHVaIm7t857bIylKJWQttDhkc=; path=/; expires=Mon, 22-Jan-24 18:02:57 GMT; domain=.api.openai.com; HttpOnly; Secure; SameSite=None","server":"cloudflare","cf-ray":"998a073ef11b70-DUB","content-encoding":"gzip","alt-svc":"h3=\":443\"; ma=86400"},"code":200,"size":"358B","time":"713ms"}
``` 

I also want to user to click on a button to copy this content in " data.value " 

Thanks again.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @quantsegu,

You can modify the transformer of the rest call if you only want the data value to be returned and accessible. Below is a screenshot of how to modify the rest transformer to do this.
<img width="967" alt="Screenshot 2024-01-23 at 08 02 55" src="https://github.com/Budibase/budibase/assets/126772285/2d4c8856-9dcb-48a9-a1c8-80497d2a95f9">

If i have miss-understood this, then just return data in transform and you still will be able to output on the value using a repeater component along with whatever additional component use wish to use.
<img width="1144" alt="Screenshot 2024-01-23 at 08 06 19" src="https://github.com/Budibase/budibase/assets/126772285/c29e3b30-f915-4b46-ad0a-03af3f2d5725">
<img width="1402" alt="Screenshot 2024-01-23 at 08 06 00" src="https://github.com/Budibase/budibase/assets/126772285/570e52e4-b953-45ae-976c-5ba77dd71c55">

I don't believe we have an action to copy to clip board, I'd recommend creating a feature request for this [here](https://github.com/Budibase/budibase/issues/new/choose).

As for the assigning things to state you can do this on a button click and the [update state](https://docs.budibase.com/docs/data-actions#update-state) action. Simply set the value of the state using repeater value shown above. Then you should be able to use this newly created state value elsewhere.

Hopefully this helps!

**Reply by quantsegu** [reply]
Hi 

Thanks for sharing the details. I did do the following transformation as part of action but some how it doesn't come well. Let me try again with what you suggested.



```javascript
const result = [];
const text =  data.choices[0].message.content;
const vl = text.value;
return text;
```

I will create a request for "Copy" command.

Regards
Lakshmi Segu
